contracts/helpers/TransferHelper.sol

```diff
@@ -324,36 +215,79 @@ contract TransferHelper is
-            // "bubble up" the conduit's revert reason.
-            revert ConduitErrorRevertBytes(data, conduitKey, conduit);
         } catch Error(string memory reason) {
             // Catch reverts with a provided reason string and
             // revert with the reason, conduit key and conduit address.
             revert ConduitErrorRevertString(reason, conduitKey, conduit);
+        } catch (bytes memory data) {
+            // Conduits will throw a custom error when attempting to transfer
+            // native token item types or an ERC721 item amount other than 1.
+            // Bubble up these custom errors when encountered. Note that the
+            // conduit itself will bubble up revert reasons from transfers as
+            // well, meaning that these errors are not necessarily indicative of
+            // an issue with the item type or amount in cases where the same
+            // custom error signature is encountered during a conduit transfer.
+
+            // Set initial value of first four bytes of revert data to the mask.
+            bytes4 customErrorSelector = bytes4(0xffffffff);
+
+            // Utilize assembly to read first four bytes (if present) directly.
+            assembly {
+                // Combine original mask with first four bytes of revert data.
+                customErrorSelector := and(
+                    mload(add(data, 0x20)), // Data begins after length offset.
+                    customErrorSelector
+                )
+            }
+
+            // Pass through the custom error in question if the revert data is
+            // the correct length and matches an expected custom error selector.
+            if (
+                data.length == 4 &&
+                (customErrorSelector == InvalidItemType.selector ||
+                    customErrorSelector == InvalidERC721TransferAmount.selector)
+            ) {
+                // "Bubble up" the revert reason.
+                assembly {
+                    revert(add(data, 0x20), 0x04)
+                }
+            }
+
+            // Catch all other reverts from the external call to the conduit and
+            // include the conduit's raw revert reason as a data argument to a
+            // new custom error.
+            revert ConduitErrorRevertBytes(data, conduitKey, conduit);
         }
     }
 
     /**
      * @notice An internal function to check if a recipient address implements
-     *         onERC721Received for a given tokenId.
+     *         onERC721Received for a given tokenId. Note that this check does
+     *         not adhere to the safe transfer specification and is only meant
+     *         to provide an additional layer of assurance that the recipient
+     *         can receive the tokens — any hooks or post-transfer checks will
+     *         fail and the caller will be the transfer helper rather than the
+     *         ERC721 contract.
```

Comment:
```suggestion
     *         conduit.
```
Perhaps add something along the lines of "operator is faked to be the conduit".

https://eips.ethereum.org/EIPS/eip-721
> /// @param _operator The address which called `safeTransferFrom` function

Reply:
Ah no, the caller is the 721 contract, my bad.

Reply:
done